### ISSUE TITLE
Question about shap values and possible cross-validation scheme

### ISSUE THREAD

**Issue by juanramonua** [shap/shap#1008]
Usually the examples with SHAP follows these steps:
1. use an X_train set to train a model (e.g. XGBoost);
2. calculate the explainer and shap values with an X_test set;
3. the shap values are analyzed.... (importance values, force_plot, etc.)

If we have few samples (e.g. 50 or 100 in the health area are usual) would it be possible to use a cross validation scheme (e.g. 10-CV) to calculate the shap values? Not to use X (the whole set) to train and to calculate the shap values. The scheme would be as follows: 
1. X is divided into 10 X_train and 10 X_test;
2. each train set is used to train a model;
3. calculate 10 explainers and 10 sets of shap values with the 10 X_test;
4. concatenate the 10 sets of shap values to get only 1.

This way we would have shap values calculated exclusively with X_test samples using trained models with their respective X_train for the whole data set.

Now, to calculate some plots we need the `explainer.expected_value` as we have 10. Would the average be calculated to make the plots? or does the proposed scheme not make sense?

Regards,

**Comment by L-Ramos** [comment]
I know this is an old unanswered issue: I proposed a simple approach for it: https://medium.com/@lucasramos_34338/visualizing-variable-importance-using-shap-and-cross-validation-bd5075e9063a

**Comment by federicocampo** [comment]
Are there some updates for this question? Or the code proposed by @L-Ramos in his article is the only meaningful way to approach it?

**Comment by pecto2020** [comment]
I like the solution of L-Ramos. It's not clear, however, if it is OK to concatenate SHAP values generated using different models (with different optimal hyperparameters). Moreover, what if each test set has a different baseline SHAP value?
